src/ws/cockpitsshtransport.c

```diff
@@ -333,170 +359,273 @@ verify_knownhost (CockpitSshData *data)
 }
 
 static const gchar *
-cockpit_ssh_authenticate (CockpitSshData *data)
+auth_result_string (int rc)
+{
+  switch (rc)
+    {
+    case SSH_AUTH_SUCCESS:
+      return "succeeded";
+    case SSH_AUTH_DENIED:
+      return "denied";
+    case SSH_AUTH_PARTIAL:
+      return "partial";
+      break;
+    case SSH_AUTH_AGAIN:
+      return "again";
+    default:
+      return "error";
+    }
+}
+
+static int
+do_password_auth (CockpitSshData *data)
 {
-  gss_cred_id_t gsscreds = GSS_C_NO_CREDENTIAL;
-  const gchar *password = NULL;
-  const gchar *problem;
-  gboolean tried = FALSE;
-  gchar *description;
   const gchar *msg;
-  int methods;
+  const gchar *password = NULL;
   int rc;
 
-  problem = "authentication-failed";
-
-  rc = ssh_userauth_none (data->session, NULL);
-  if (rc == SSH_AUTH_ERROR)
+  password = cockpit_creds_get_password (data->creds);
+  rc = ssh_userauth_password (data->session, NULL, password);
```

Comment:
If password is supposed to be non-null (based on a condition in a calling function in the same file) then we should assert it.